app/src/main/java/com/severianfw/picto/viewmodel/HomeViewModel.kt

```diff
@@ -20,45 +19,47 @@ class HomeViewModel @Inject constructor(
 ) : ViewModel() {
 
     private val _photos = MutableLiveData<List<ImageUrl>>()
-    val photos: LiveData<List<ImageUrl>> = _photos
+    val photos: LiveData<List<ImageUrl>> get() = _photos
 
     private val _isLoading = MutableLiveData<Boolean>()
-    val isLoading: LiveData<Boolean> = _isLoading
+    val isLoading: LiveData<Boolean> get() = _isLoading
 
-    private var _pageNum: Int = 1
+    private val _isResponseSuccess = MutableLiveData<Boolean>()
+    val isResponseSuccess: LiveData<Boolean> get() = _isResponseSuccess
 
-    private val compositeDisposable = CompositeDisposable()
-    private val newPhotos = mutableListOf<ImageUrl>()
+    var isInitial: Boolean = false
 
-    init {
-        getPhotos()
-    }
+    private var pageNumber: Int = 1
+
+    private val compositeDisposable = CompositeDisposable()
+    private val newPhoto = mutableListOf<ImageUrl>()
 
-    private fun getPhotos() {
+    fun getPhotos() {
         _isLoading.value = true
         compositeDisposable.add(
-            getPhotoUseCase(_pageNum).subscribeOn(Schedulers.io())
+            getPhotoUseCase(pageNumber).subscribeOn(Schedulers.io())
                 .observeOn(AndroidSchedulers.mainThread())
+                .doFinally { _isLoading.value = false }
                 .subscribeWith(object : DisposableSingleObserver<List<PhotoResponse>>() {
-                    override fun onSuccess(t: List<PhotoResponse>) {
-                        for (item in t) {
-                            item.urls?.let { newPhotos.add(it) }
+                    override fun onSuccess(response: List<PhotoResponse>) {
+                        for (item in response) {
+                            item.urls?.let { newPhoto.add(it) }
```

Comment:
I think, no need to check null or not, because it will alter the size of the response and data shown to user
Please check the previous comment @severianfw